### PR TITLE
#10208 remove unnecessary queries

### LINKED ISSUE #10208: Remove unnecessary queries for embedded entities [3]
### Problem Description
When saving a core dto, like a CaseDataDto, the system calls the fromDto method for all sub entities (e.g. SymptomsDto), which executes a service.getByUuid methode to get the entity. This is unnecessary since the entity is already fetched as part of the core entity. Overall it costs quite some performance (https://github.com/hzi-braunschweig/SORMAS-Project/issues/3481#issuecomment-773260588):

> fillOrBuildEntity set hospital/therapy/clinical/maternal/portHealth: 67ms
> fillOrBuildEntity setSymptoms: 12ms

### Proposed Change
Replace the fromDto method in the related facade with a fillOrBuildEntity method that also takes the entity as a parameter.

Note: The "OrBuild" part is not needed here. I'd just do it, to keep it simple and consistent with existing logic like CaseFacadeEjb.fillOrBuildEntity.

Do this for the following methods:

**Important:**

* **When calling this from a freshly created core dto, the uuid needs to be overridden based on the value from the dto. When it's an existing one, the uuid should be validated.**
* **When calling this from a freshly created dto, the system will no longer look for existing sub-entities (e.g. symptoms entity of a case) and instead expect they they also have to be created. This does of course not go for reference dtos.**

- [x] SymptomsFacadeEjb.fromDto
- [x] HospitalizationFacadeEjb.fromDto
- [x] HospitalizationFacadeEjb.fromDto for PreviousHospitalization
- [x] EpiDataFacadeEjb.fromDto
- [x] TherapyFacadeEjb.fromDto
- [x] ClinicalCourseFacadeEjb.fromDto
- [x] MaternalHistoryFacadeEjb.fromDto
- [x] PortHealthInfoFacadeEjb.fromDto
- [x] HealthConditionsMapper.fromDto (just needs to be removed, fillOrBuild already exists).
- [x] SormasToSormasOriginInfoFacadeEjb.fromDto
- [x] LocationFacadeEjb.fromDto

I'd suggest to do the same for the following methods to keep it consistent and error prove for the future:

- [x] TaskFacadeEjb.fromDto
- [x] ActionFacadeEjb.fromDto
- [x] DocumentFacadeEjb.fromDto
- [x] DiseaseConfigurationFacadeEjb.fromDto
- [x] CampaignFormDataFacadeEjb.fromDto
- [x] CampaignDiagramDefinitionFacadeEjb.fromDto
- [x] CampaignFormMetaFacadeEjb.fromDto
- [x] ClinicalVisitFacadeEjb.fromDto
- [x] EventGroupFacadeEjb.fromDto
- [x] ExternalMessageFacadeEjb.fromDto
- [x] TestReportFacadeEjb.fromDto
- [x] FeatureConfigurationFacadeEjb.fromDto
- [x] PopulationDataFacadeEjb.fromDto
- [x] OutbreakFacadeEjb.fromDto
- [x] AggregateFacadeEjb.fromDto
- [x] WeeklyReportFacadeEjb.fromDto
- [x] PathogenTestFacadeEjb.fromDto
- [x] AdditionalTestFacadeEjb.fromDto
- [x] SampleFacadeEjb.fromDto
- [x] SormasToSormasShareRequestFacadeEjb.fromDto
- [x] SystemEventFacadeEjb.fromDto
- [x] PrescriptionFacadeEjb.fromDto
- [x] TreatmentFacadeEjb.fromDto
- [x] UserFacadeEjb.fromDto
- [x] UserRoleFacadeEjb.fromDto
- [x] VaccinationFacadeEjb.fromDto (just needs to be removed, fillOrBuild already exists).
- [x] VisitFacadeEjb.fromDto

**Addition**

- [x] Make sure the related child entities for cases and contacts are fetched when saving or loading the whole entity (as they are mostly lazy)

### PR DESCRIPTION
**Pull request by razvancornita**
Fixes #10208

### CODE REVIEW

**Comment by razvancornita** [comment]
Tests failing: 
- de.symeda.sormas.backend.caze.CaseFacadeEjbTest#testMergeCase
- de.symeda.sormas.backend.caze.CaseFacadeEjbTest#testCloneCaseActivityAsCaseIsCloned
- de.symeda.sormas.backend.common.StartupShutdownServiceTest#testHistoryTablesMatch
- de.symeda.sormas.backend.deletionconfiguration.CoreEntityDeletionServiceTest#testCaseVisitAutomaticDeletion
- de.symeda.sormas.backend.epidata.EpiDataPseudonymizationTest#getEpiDataInJurisdiction
- de.symeda.sormas.backend.epidata.EpiDataPseudonymizationTest#getEpiDataOutsideJurisdiction
- de.symeda.sormas.backend.sample.PathogenTestFacadeEjbTest#testSaveAndUpdatePathogenTestAssociatedToBothCaseAndContact
- de.symeda.sormas.backend.sample.SampleFacadeEjbTest#testGetIndexListForCaseConvertedFromContact
- de.symeda.sormas.backend.sormastosormas.entities.SormasToSormasCaseFacadeEjbTest#testSaveSharedCaseWithInfrastructureExternalId
- de.symeda.sormas.backend.sormastosormas.entities.SormasToSormasCaseFacadeEjbTest#testReportingUserIsIncludedButUpdated
- de.symeda.sormas.backend.sormastosormas.entities.SormasToSormasCaseFacadeEjbTest#testSaveSharedCaseWithInfrastructureName
- de.symeda.sormas.backend.sormastosormas.entities.SormasToSormasContactFacadeEjbTest#testReportingUserIsIncludedButUpdated
- de.symeda.sormas.backend.sormastosormas.entities.SormasToSormasEventFacadeEjbTest#testSyncRecursively
- de.symeda.sormas.backend.sormastosormas.entities.SormasToSormasEventFacadeEjbTest#testReportingUserIsIncludedButUpdated 
- de.symeda.sormas.backend.symptoms.SymptomsServiceTest#testCaseVisitAutomaticDeletion
- de.symeda.sormas.backend.vaccination.VaccinationFacadeEjbTest#testSave

File: sormas-api/src/main/java/de/symeda/sormas/api/uuid/MismatchUuidException.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+@SuppressWarnings("serial")
+public class MismatchUuidException extends ValidationRuntimeException {
+	public MismatchUuidException(String entityUuid, Class entityClass, String dtoUuid) {
+		super(
+				String.format(
+						"%s dto UUID (%s) does not match entity UUID (%s)",
+						entityClass.getSimpleName(),
+						dtoUuid,
+						entityUuid));
```
Comment (ChristopherRiedel) [inline sormas-api/src/main/java/de/symeda/sormas/api/uuid/MismatchUuidException.java:30]: Does it just look like that, or is the formatting wrong here?

File: sormas-backend/src/main/java/de/symeda/sormas/backend/sormastosormas/share/incoming/SormasToSormasShareRequestFacadeEJB.java [hunk 2]
```diff
@@ -251,14 +256,14 @@ public ShareRequestDetailsDto getShareRequestDetails(String uuid) {
...
-		SormasToSormasShareRequest target =
-			DtoHelper.fillOrBuildEntity(source, shareRequestService.getByUuid(source.getUuid()), SormasToSormasShareRequest::new, checkChangeDate);
+	private SormasToSormasShareRequest fillOrBuildEntity(@NotNull SormasToSormasShareRequestDto source, SormasToSormasShareRequest target, boolean checkChangeDate) {
+		target = DtoHelper.fillOrBuildEntity(source, target, SormasToSormasShareRequest::new, checkChangeDate);
 
 		target.setDataType(source.getDataType());
 		target.setStatus(source.getStatus());
-		target.setOriginInfo(originInfoFacade.fromDto(source.getOriginInfo(), checkChangeDate));
+
+		//TODO: originInfo should be reference type
```
Comment (ChristopherRiedel) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/sormastosormas/share/incoming/SormasToSormasShareRequestFacadeEJB.java:265]: What's about this TODO?

File: sormas-api/src/main/java/de/symeda/sormas/api/uuid/MismatchUuidException.java (hunk 1 above)
Comment (MartinWahnschaffe) [reply sormas-api/src/main/java/de/symeda/sormas/api/uuid/MismatchUuidException.java:30]: I guess so.
@razvancornita Can you please take care of this? Make sure your IDE is configured as described here:
https://github.com/hzi-braunschweig/SORMAS-Project/blob/development/docs/DEVELOPMENT_ENVIRONMENT.md#step-5-configure-code-formatting-and-import-settings

**Review by ChristopherRiedel — APPROVED** [review]

File: sormas-backend/src/main/java/de/symeda/sormas/backend/externalmessage/labmessage/SampleReportFacadeEjb.java [hunk 3]
```diff
@@ -90,7 +90,7 @@ public SampleReport fromDto(@NotNull SampleReportDto source, ExternalMessage lab
 		if (source.getTestReports() != null) {
 			List<TestReport> testReports = new ArrayList<>();
 			for (TestReportDto t : source.getTestReports()) {
-				TestReport testReport = testReportFacade.fromDto(t, target, false);
+				TestReport testReport = testReportFacade.fromDto(t, false);
```
Comment (FredrikSchaeferVitagroup) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/externalmessage/labmessage/SampleReportFacadeEjb.java:93]: This change breaks the ability to save detailed lab messages when importing them from the SORMAS-DEMIS-adapter.

**Pull request merged**